init: holochain-go at 0.1.0-alpha

gavinrogers opened the pull request:
Holochain is a platform for creating an unenclosable carrier upon which
highly-scalable dApps and smart contracts can be built. This is the holochain
team's initial implimentation in go. A rust version is under development.

###### Motivation for this change

Add the holochain core to nixpkgs to allow app developers to bootstrap more quickly: allow people to run their own holochain node.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
init: holochain-go at 0.1.0-alpha

Holochain is a platform for creating an unenclosable carrier upon which
highly-scalable dApps and smart contracts can be built. This is the holochain
team's initial implimentation in go. A rust version is under development.

dtzWill committed:
tor-browser-bundle-bin: 7.5.5 -> 7.5.6

mnacamura committed:
feedgnuplot: init at 1.49

matthewbauer committed:
generators: add PLIST handling

/cc @LnL7 @3noch

matthewbauer committed:
xcbuild: rework to use lib.PLIST

This is a good way to test that plist works.

Sorry for the large diff. There are a bunch of cleanups in here that
needed to be done. Should make it possible to use in cross
compilation.

matthewbauer committed:
unixtools: add killall

matthewbauer committed:
treewide: use killall

Commit:
iperf2: 2.0.11 -> 2.0.12

Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools.

This update was made based on information from https://repology.org/metapackage/iperf/versions.

These checks were done:

- built on NixOS
- /nix/store/c7p3b3b353pssma0cg1wg78pfs6x2is3-iperf-2.0.12/bin/iperf passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 0 of 1 passed binary check by having the new version present in output.
- found 2.0.12 with grep in /nix/store/c7p3b3b353pssma0cg1wg78pfs6x2is3-iperf-2.0.12
- directory tree listing: https://gist.github.com/457110ea6db3cd3dbd6c9bc7ff82477a
- du listing: https://gist.github.com/86655ffd63846a5d3a473393af8e4da9

mnacamura committed:
feedgnuplot: install shell completions

Commit:
eid-mw: 4.4.2 -> 4.4.3

Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools.

This update was made based on information from https://repology.org/metapackage/eid-mw/versions.

These checks were done:

- built on NixOS
- Warning: no invocation of /nix/store/sis599r39a2g7kvnv1d09l6sy6kn45l0-eid-mw-4.4.3/bin/eid-viewer had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/sis599r39a2g7kvnv1d09l6sy6kn45l0-eid-mw-4.4.3/bin/.eid-viewer-wrapped had a zero exit code or showed the expected version
- /nix/store/sis599r39a2g7kvnv1d09l6sy6kn45l0-eid-mw-4.4.3/bin/beid-update-nssdb passed the binary check.
- /nix/store/sis599r39a2g7kvnv1d09l6sy6kn45l0-eid-mw-4.4.3/bin/eid-nssdb passed the binary check.
- 2 of 4 passed binary check by having a zero exit code.
- 0 of 4 passed binary check by having the new version present in output.
- found 4.4.3 with grep in /nix/store/sis599r39a2g7kvnv1d09l6sy6kn45l0-eid-mw-4.4.3
- directory tree listing: https://gist.github.com/3bca8b9d321e62009daf1d388923ec47
- du listing: https://gist.github.com/64220a7b5b960cbd2ec3bb5f61c93a29

Commit:
deadbeef-mpris2-plugin: 1.10 -> 1.11

Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools.

This update was made based on information from https://repology.org/metapackage/deadbeef-mpris2-plugin/versions.

These checks were done:

- built on NixOS

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 1.11 with grep in /nix/store/isa0i2bjry1xg3l8895k10w1zwd4ij9f-deadbeef-mpris2-plugin-1.11
- directory tree listing: https://gist.github.com/020d669dc365b545d61d171c5566d03a
- du listing: https://gist.github.com/9cd938a3d8215dbc4f00d62fe20ca4f5

mnacamura committed:
feedgnuplot: disable tests

dtzWill committed:
lit: 0.5.1 -> 0.6.0

matthewbauer committed:
libcCrossChooser: fix xcode eval

it should be targetPackages.darwin.iosSdkPkgs.libraries

matthewbauer committed:
xcode: default to 8.2

There is some weird issue with 9.4 that needs to be resolved. 8.2 is
used by iOS anyway so that makes it easier.

matthewbauer committed:
generators: refactor toPLIST

matthewbauer committed:
generators: refactor toPlist

Address PR comments

Refactors

- Rename toPLIST -> toPlist

r-ryantm committed:
gdcm: 2.8.6 -> 2.8.7 (#42708)

Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools.

This update was made based on information from https://repology.org/metapackage/gdcm/versions.

These checks were done:

- built on NixOS
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcm2vtk passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmviewer passed the binary check.
- Warning: no invocation of /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcm2pnm had a zero exit code or showed the expected version
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmdump passed the binary check.
- Warning: no invocation of /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmdiff had a zero exit code or showed the expected version
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmraw passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmscanner passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmanon passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmgendir passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmimg passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmconv passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmtar passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcminfo passed the binary check.
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmscu passed the binary check.
- Warning: no invocation of /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmxml had a zero exit code or showed the expected version
- /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7/bin/gdcmpap3 passed the binary check.
- 13 of 16 passed binary check by having a zero exit code.
- 0 of 16 passed binary check by having the new version present in output.
- found 2.8.7 with grep in /nix/store/b5k2wcykj6s3j1rax851vpzyj6amqpvk-gdcm-2.8.7
- directory tree listing: https://gist.github.com/d4d202ac57e29b5fe6d46f1e8b7b112b
- du listing: https://gist.github.com/7244757ac1449adb1adde3c480e77c8d

infinisil committed:
treewide: http -> https sources (#42676)

* treewide: http -> https sources

This updates the source urls of all top-level packages from http to
https where possible.

* buildtorrent: fix url and tab -> spaces

Mic92 committed:
lxd: also needs ebtables

matthewbauer committed:
libcCross: Use correct libSystem

Anton-Latukha committed:
mbedtls: cmake&ninja, threading, clean-up; hiawatha: options, Nix mbedtls (#41722)

* mbedtls: build with cmake&ninja, clean-up

* mbedtls: cmake ninja Darwin build clean-up

* hiawatha: add build options, use system mbedTLS, platforms -> unix

Mic92 commented on pkgs/servers/holochain-go/default.nix:
```diff
@@ -0,0 +1,31 @@
+  src = fetchgit {
+    inherit rev;
+    url = "https://github.com/holochain/holochain-proto";
+    sha256 = "19l29jnr63ximmyn4i4llv2mdwh306c2mpzmx2anj9z12wjpach0";
```
Can you use `fetchFromGitHub` since you are not using git submodules?

gavinrogers replied:
Yup, done

Mic92 commented on pkgs/servers/holochain-go/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  goDeps = ./deps.nix;
+
+  # TODO: add metadata https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes
```
this one can be removed.

gavinrogers replied:
Done

Mic92 commented on pkgs/servers/holochain-go/default.nix:
```diff
@@ -0,0 +1,31 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.2.1
```
Since this was manually edited, we should not claim it was generated.

gavinrogers replied:
Ah ok that makes sense!

Mic92 commented on pkgs/servers/holochain-go/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+buildGoPackage rec {
+  name = "holochain-go${version}";
+  version = "v0.1.0-alpha";
```
we usually do not include the `v` in the name, but rather do: `holochain-go-0.1.0-alpha`:

```nix
nix-repl> builtins.parseDrvName "holochain-gov0.1.0-alpha"
{ name = "holochain-gov0.1.0-alpha"; version = ""; }
nix-repl> builtins.parseDrvName "holochain-go-0.1.0-alpha"
{ name = "holochain-go"; version = "0.1.0-alpha"; }
```

gavinrogers replied:
Thanks!

svanderburg committed:
nodePackages: regenerate with node2nix 1.6.0

primeos committed:
lf: 5 -> 6

Commit:
Cleanup: remove autogenerated line

Agree with @Mic92: "Since this was manually edited, we should not claim it was
auto generated.

Commit:
Cleanup: remove line left over from template

As per @Mic92: can be removed

Commit:
Cleanup: remove v from version and use fetchFromGithub

More housecleaning, also as per @Mic92. He's right, as it says in the manual, to
use fetchFromGithub instead of fetchgit

> Best: Fetches a snapshot archive and you get the rev you want.

Commit:
Fix: Capitalisation of downloadPage

Profpatsch reviewed (changes requested):
You made a mistake pushing the last changes.

gavinrogers reviewed (commented):
Thanks for the very helpful review @Mic92, I've brought all these points into line. I somehow ended up getting some other code into this code review, I guess I shouldn't rebase onto master when doing code cleanup, as the PR stays on the same rev?

gavinrogers commented:
#42937 Is what I've opened to replace this @Mic92  I've incorporated your feedback into commits in that PR
